### ISSUE TITLE
[trainer] Add gradient accumulation option

### ISSUE THREAD

**Issue by frgfm** [frgfm/Holocron#235]
All trainers should have an option to use gradient accumulation so that we can perform training at decent batch size without a beefy GPU.
